nixos/modules/services/network-filesystems/ipfs-cluster.nix

```diff
@@ -0,0 +1,136 @@
+{
+  config,
+  lib,
+  pkgs,
+  options,
+  ...
+}:
+let
+  cfg = config.services.ipfs-cluster;
+
+  # secret is by envvar, not flag
+  initFlags = toString [
+    (lib.optionalString (cfg.initPeers != [ ]) "--peers")
+    (lib.strings.concatStringsSep "," cfg.initPeers)
+  ];
+in
+{
+  options = {
+
+    services.ipfs-cluster = {
+
+      enable = lib.mkEnableOption "Pinset orchestration for IPFS - requires ipfs daemon to be useful";
+
+      user = lib.mkOption {
```

Comment:
Don't you also need to configure `users.users.ipfs` and `user.groups.ipfs` below? Most other modules do that.

Reply:
Oh, right. In that case it would be good to add an assert that the users are equal, right?

Reply:
There's also an assertion for this
```nix
    assertions = [ 
      {   
        assertion = cfg.enable -> config.services.kubo.enable;
        message = "ipfs-cluster requires ipfs - configure and enable services.kubo";
      }   
    ];
```

Reply:
Sounds good, adding
```nix
      {   
        assertion = cfg.user == config.services.kubo.user;
        message = "services.ipfs-cluster.user must match services.kubo.user";
      }   
      {   
        assertion = cfg.group == config.services.kubo.group;
        message = "services.ipfs-cluster.group must match services.kubo.group";
      } 
```

Reply:
Yes, although in that case you could just remove `cfg.user` / `cfg.group` and use services.kubo.* directly, I think?